### PR TITLE
feat: use custom capability_type for the CPT

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Registers the prompt CPT with a custom `capability_type`, in order to allow granular control over access to this CPT.

### How to test the changes in this Pull Request:

1. There should be no functional changes, verify that prompts are editable and visible on the front-end.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/class-newspack-popups.php
Comment (leogermani) [inline includes/class-newspack-popups.php:189]: nit, but you don't need to redo this. You can:

```
		$args = [
			'map_meta_cap'    => true,
			'capability_type' => self::NEWSPACK_POPUPS_CPT,
			'capabilities'    => []
		];
		
		return get_post_type_capabilities( (object) $args );
```

Actually `$args` are the same arguments passed to register the post type. So if you had that array available, you could use the same array in both places.

File: includes/class-newspack-popups.php
Comment (leogermani) [inline includes/class-newspack-popups.php:82]: This only needs to run once, the roles capabilities are stored in the DB. We don't want to run this on every execution

File: includes/class-newspack-popups.php
Comment (adekbadek) [reply includes/class-newspack-popups.php:194]: Thanks! Added in a55dc11c468c03b23dfa012ee4d3ba2144a7c646

File: includes/class-newspack-popups.php
Comment (adekbadek) [reply includes/class-newspack-popups.php:82]: Fixed in 39ba1b9765909ced595a0c675adbcb602f00eaa3. The recommended way to set up caps on plugin activation, but the plugin is already active on sites.

**Comment by adekbadek** [comment]
Pinging @leogermani – I'd appreciate a re-review since the corresponding changes in `newspack-plugin` will soon be released 🙏

**Review by laurelfulford — CHANGES_REQUESTED** [review]
Hi @adekbadek! I ran a review on the functionality while waiting for Leo's approval for the code changes -- with this PR and the associated Newspack Plugin one, I:

* Made a role with the capabilities to access `newspack wizard newspack-popups-wizard` plus the post and admin settings you suggested, and all the Newspack Campaign CPTs. Assigned this role, I could successfully edit/add campaigns and prompts, and make changes to the Segments and Settings tabs.
* I made a role that had no access to campaigns, and when assigned this role successfully... didn't see anything to do with campaigns 😆 
* I tested these PR changes with PublishPress Capabilities disabled and repeated the steps in the first bullet point, and it seemed fine.

The only issue I ran into is when I tried mixing up the campaign CPT capabilities -- like when I removed the ability to delete the popup CPTs, it removed the 'Move to trash' button from the prompt editor, but it didn't remove the 'Delete' option for the prompt from the main campaigns screen, or stop me from deleting one:

Settings:
image

Individual prompt post: 
image

Options under campaigns wizard: 
image

Something similar happens when you restrict editing access (you can still pick edit from the dropdown from the wizard), but then you get a WordPress error, so I think it's technically OK:

image

Just let me know if you have any questions about this!

**Comment by adekbadek** [comment]
> like when I removed the ability to delete the popup CPTs, it removed the 'Move to trash' button from the prompt editor, but it didn't remove the 'Delete' option for the prompt from the main campaigns screen, or stop me from deleting one:

This is actually by design. The idea is that the wizard capabilities should override the CPT capabilities – you get access to the whole wizard, not just parts of it. 

I don't think that such fine-grained permissions are justifiable at this point. I imagine a publisher granting an employee capability to manage prompts, period. A scenario where an employee is allowed to e.g. edit prompts but not publish or delete them is _imaginable_, but I don't think it'd be common. We can revisit if we get feedback requesting such fine-grained control over the wizard.

**Review by leogermani — APPROVED** [review]

**Review by laurelfulford — APPROVED** [review]
> We can revisit if we get feedback requesting such fine-grained control in the wizard.

That all makes sense -- thanks @adekbadek!

**Comment by laurelfulford** [comment]
Going to merge this so I can spin up fresh alphas! 🙂

**Pull request merged**
